h2o-algos/src/main/java/hex/svd/SVD.java

```diff
@@ -701,9 +706,12 @@ public void computeImpl() {
           }
 
           for (int k = 1; k < _parms._nv; k++) {  // loop through for each eigenvalue/eigenvector...
-            if (stop_requested()) break;
-            if (_matrixRankReached) { // number of eigenvalues found is less than _nv
-              int newk = k-1;
+            if (_matrixRankReached || stop_requested()) { // number of eigenvalues found is less than _nv
+              if (stop_requested()) {
+                _matrixRankReached = true;
```

Comment:
Thanks Wendy.

The issue I see with setting the parameter to true is that it doesn't actually capture the property of the dataset: we don't now if it is rank deficient, we stopped early because of the timeout. Setting this parameter to true is therefore a little confusing.

If I read the code correctly the only part that depends on it is this:

        if (_matrixRankReached) { // need to shorten the correct eigen stuff
          model._output._d = Arrays.copyOf(model._output._d, _parms._nv);
          for (int index=0; index < model._output._v.length; index++) {
            model._output._v[index] = Arrays.copyOf(model._output._v[index],  _parms._nv);
          }
        }

Can we change it simply to

        if (_parms._nv < model._output._d.length) { // need to shorten the correct eigen stuff
          model._output._d = Arrays.copyOf(model._output._d, _parms._nv);
          for (int index=0; index < model._output._v.length; index++) {
            model._output._v[index] = Arrays.copyOf(model._output._v[index],  _parms._nv);
          }
        }

and keep the "_matrixRankReached" unmodified.

We can also change the name of "_matrixRankReached" to something else. My point is that it should correctly describe the state of things. If we stop early the rank was not actually reached.